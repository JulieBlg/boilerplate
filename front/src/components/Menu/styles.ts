import { Theme } from '@material-ui/core';

const styles = (theme: Theme) => ({
  drawer: {
    minWidth: theme.spacing(24),
    padding: 0,
  },
});

export default styles;
