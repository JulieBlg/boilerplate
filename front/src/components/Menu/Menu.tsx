import React, { FC } from 'react';
import Drawer from '@material-ui/core/Drawer';
import Divider from '@material-ui/core/Divider';
import IconButton from '@material-ui/core/IconButton';
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
import ChevronRightIcon from '@material-ui/icons/ChevronRight';
import Toolbar from '@material-ui/core/Toolbar';
import { useTheme, makeStyles } from '@material-ui/core/styles';

import styles from './styles';

interface Props {
  open: boolean;
  handleDrawerClose: () => void;
}

const Menu: FC<Props> = ({ open, handleDrawerClose }) => {
  const theme = useTheme();
  const classes = makeStyles(styles)();

  return (
    <Drawer variant="persistent" anchor="left" open={open}>
      <Toolbar className={classes.drawer}>
        <IconButton onClick={handleDrawerClose}>
          {theme.direction === 'ltr' ? (
            <ChevronLeftIcon />
          ) : (
            <ChevronRightIcon />
          )}
        </IconButton>
      </Toolbar>
    </Drawer>
  );
};

export default Menu;
