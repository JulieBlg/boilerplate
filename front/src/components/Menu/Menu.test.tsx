import React from 'react';
import { render } from '@testing-library/react';

import Menu from './Menu';

describe('Menu', () => {
  it('should render properly', () => {
    const { container } = render(<Menu />);

    expect(container).toMatchSnapshot();
  });
});
