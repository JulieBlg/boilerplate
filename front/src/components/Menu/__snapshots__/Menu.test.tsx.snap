// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render properly 1`] = `
<div>
  <div
    class="MuiDrawer-root MuiDrawer-docked"
  >
    <div
      class="MuiPaper-root MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
      style="visibility: hidden; webkit-transform: translateX(-0px); transform: translateX(-0px);"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular makeStyles-drawer-1 MuiToolbar-gutters"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
